test/release/examples/primers/records.chpl

```diff
@@ -0,0 +1,172 @@
+// Records
+
+//
+// This primer covers the declaration and uses of records.
+//
+
+/*
+  
+   A record is a type that can contain variables and constants,
+   called fields, as well as functions and iterators called methods.
+   Records have many similaties to classes, but there are several important
+   differences:
+  
+   * classes support inheritance and virtual dispatch while records do not
+   * different class variables can refer to the same instance,
+     while record variables refer to distinct memory
+   * copy-initialization and assignment can be implemented for records
+     but not for classes.
+  
+   A record type can be declared using the ``record`` keyword:
+*/ 
+record Color {
+  var red: uint(8);
+  var green: uint(8);
+  var blue: uint(8);
+}
+
+// The ``new`` keyword creates a new record by calling an initializer.
+// The record ``Color`` doesn't declare any initializers, and so the
+// compiler-generated one is used. The compiler-generated initializer
+// has an argument for each field in the record.
+
+// The result of ``new`` can be captured in a variable:
+var taupe = new Color(179, 139, 109);
+writeln(taupe); // output: (red = 179, green = 139, blue = 109)
+
+// Alternatively, a variable can be declared using the record type
+// without initializing it.
+// In that event the variable is default-initialized.
+// The compiler-generated initializer allows default initialization.
+// It simply default-initializes each field in turn.
+var myColor: Color;
+writeln(myColor); // output: (red = 0, green = 0, blue = 0)
+
+// Records support methods, both declared within the record
+// or declared outside of the record.
+// Note that inside of a method, the fields may be accessed
+// by name. Additionally `this` refers to the entire record.
+proc Color.luminance() {
+  return 0.2126*red + 0.7152*green + 0.0722*blue;
+}
+writeln(taupe.luminance());
+
+// Here is an example of a record to store a point where
+// a method is declared inside of the record declaration.
+// This record also demonstrates a a few initializers.
+
+record Point {
+  var x: real;
+  var y: real;
+  var z: real;
+
+  // This is a default initializer
+  proc init() {
+    writeln("Point default-init");
+    this.x = 0.0;
+    this.y = 0.0;
+    this.z = 0.0;
+  }
+
+  // This is a useful initializer for this type
+  proc init(x: real, y: real, z: real) {
+    this.x = x;
+    this.y = y;
+    this.z = z;
+  }
+
+  // This is a copy initializer
+  proc init(from: Point) {
+    writeln("Point copy-init");
+    this.x = from.x;
+    this.y = from.y;
+    this.z = from.z;
+  }
+  proc magnitude() {
+    return sqrt(x*x + y*y + z*z);
+  }
+}
+
+// Let's create a default-initialized Point
+var emptyPoint:Point; // outputs Point default-init
```

Comment:
Maybe use a colon like the other comments (e.g. "outputs: "), or wrap the output in quotes?